src/kafka-extensions/registry.ts

```diff
@@ -0,0 +1,185 @@
+
+    /**
+     * Returns the clusters managed by the provider which must be added to the kafka explorer.
+     *
+     * @param clusterSettings the cluster settings.
+     */
+    async collectClusters(clusterSettings: ClusterSettings): Promise<Cluster[] | undefined> {
+        const processor = await this.getProcessor();
+        return processor.configureClusters(clusterSettings);
+    }
+
+    /**
+     * Create the Kafka JS client instance from the given connection options.
+     *
+     * @param connectionOptions the connection options.
+     */
+    async createKafkaConfig(connectionOptions: ConnectionOptions): Promise<KafkaConfig | undefined> {
+        const processor = await this.getProcessor();
+        if (processor.createKafkaConfig) {
+            return processor.createKafkaConfig(connectionOptions);
+        }
+    }
+
+    private async getProcessor(): Promise<ClusterProviderParticipant> {
+        if (this.processor) {
+            return this.processor;
+        }
+        // The cluster provider processor is not already loaded, try to activate the owner extension.
+        // The return of the extension activate() method must return the cluster provider processor.
+        const extension = vscode.extensions.getExtension(this.extensionId);
+        if (!extension) {
+            throw new Error(`Error while getting cluster provider processor. Extension ${this.extensionId} is not available.`);
+        }
+
+        // Wait for extension is activated to get the processor
+        const result = await extension.activate();
+        if (!result) {
+            throw new Error(`Error while getting cluster provider processor. Extension ${this.extensionId}.activate() should return 'KafkaExtensionParticipant'.`);
+        }
+        if (isKafkaExtensionParticipant(result)) {
+            this.processor = (<KafkaExtensionParticipant>result).getClusterProviderParticipant(this.id);
+        }
+        if (!this.processor) {
+            throw new Error(`Error while getting cluster provider processor. Extension ${this.extensionId}.activate() should return 'KafkaExtensionParticipant'.`);
+        }
+        return this.processor;
+    }
+}
+
+function isKafkaExtensionParticipant(arg: any): arg is KafkaExtensionParticipant {
+    return (arg as KafkaExtensionParticipant).getClusterProviderParticipant !== undefined;
+}
+
+const defaultClusterProviderId = 'vscode-kafka.manual';
+
+let providers: Map<string, ClusterProvider> = new Map();
+
+export function getClusterProvider(clusterProviderId?: string): ClusterProvider | undefined {
+    intializeIfNeeded();
+    return providers.get(clusterProviderId || defaultClusterProviderId);
+}
+
+export function getClusterProviders(): ClusterProvider[] {
+    intializeIfNeeded();
+    // "Configure manually" provider must be the first
+    const manual = getClusterProvider(defaultClusterProviderId);
+    // Other providers must be sorted by name ascending
+    const others = [...providers.values()]

```

Comment:
simply don't load the default cluster provider with the others

Reply:
or ensure the default cluster provider is inserted in the 1st place, in collectClusterProviderDefinitions